### PR TITLE
QueryEditorRow: Show query errors next to query in a consistent way across Grafana

### PR DESCRIPTION
**Pull request by torkelo**
Problem
* In panel edit there is no easy way to view query errors other than hovering over the small red icon in the upper left corner of the panel 
* If you have many queries there is no way to know which query caused error 
* Inconsistency: in explore we show Alert box (below all queries). So we show more prominently but still no way to correlate which query caused it 

Figma (contains screenshot of problem definition, and exploration of alternative options) 
 https://www.figma.com/file/uYcpLusmEsMpRLwzaXtHfS/Query-errors?node-id=1669%3A14191

Solution
* Show query error below each query (when we have errors that have query refId) 
* Remove showing errors that have query id below all queries in explore (So we do not show them double)

Screenshot from 2022-04-12 11-30-04

Example with builder
Screenshot from 2022-04-12 11-30-43

Example from explore with SQL editor and larger error 

Screenshot from 2022-04-12 11-26-43

### CODE REVIEW

**Comment by dprokop** [comment]
@torkelo think this works great. Pushed minor UI change - changed the icon type and spacing. Think exclamation marg in triangle looks worse than in the circle ;)

image

Feel free to revert that commit if you feel it doesn't work better.

**Comment by torkelo** [comment]
Yea circle is better!

File: public/app/features/query/components/QueryEditorRow.tsx [hunk 1]
```diff
@@ -419,6 +421,7 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
               )}
               {editor}
             </ErrorBoundaryAlert>
+            {data && data.error && data.error.refId === query.refId && <QueryErrorAlert error={data.error} />}
```
Comment (dprokop) [inline public/app/features/query/components/QueryEditorRow.tsx:423]: Can you extract `data && data.error && data.error.refId === query.refId` to a variable?

**Review by dprokop — APPROVED** [review]
LGTM! Nice improvement

**Comment by amy-super** [comment]
@torkelo I really like this improvement. I'm worried about the contrast though - the red/pink of the error on the secondary background fails contrast checks unless the text is larger than 18 pt. Is it possible to use a darker background or do something to make sure the contrast is high enough? I do like the option with the smaller alert style but agree that's a bigger discussion about the component.

**Comment by torkelo** [comment]
@amy-super the contrast is 4.97 which is ok for AA , it is the same contrast as our secondary text on the secondary background. 

So think to address this we have to tweak the theme colors. 

Using primary background with just a border is also an option: 
Screenshot from 2022-04-12 15-16-28
Screenshot from 2022-04-12 15-18-35

**Comment by torkelo** [comment]
@staton-hyse11 @diegoadams  @amy-super 

Option A) 
Screenshot from 2022-04-14 09-55-51

Option B) 
Screenshot from 2022-04-14 09-52-31

Option C)
Screenshot from 2022-04-14 09-50-49

Don't honestly have much of a preference. I know adding a border to our current Alert box was something explored in the design system, I opened a PR to try it but I got no response/review from anyone in UX so I closed it. https://github.com/grafana/grafana/pull/37729  (not sure the border is good/bad).

**Comment by torkelo** [comment]
@staton-hyse11 @diegoadams  @amy-super 

Option A) 
Screenshot from 2022-04-14 09-55-51

Option B) 
Screenshot from 2022-04-14 09-52-31

Option C)
Screenshot from 2022-04-14 09-50-49

Don't honestly have much of a preference. I know adding a border to our current Alert box was something explored in the design system, I opened a PR to try it but I got no response/review from anyone in UX so I closed it. https://github.com/grafana/grafana/pull/37729  (not sure the border is good/bad).

**Comment by diegoadams** [comment]
> Option B) Screenshot from 2022-04-14 09-52-31
> 


My vote is for Option B as it aligns more with the Alert box and keeps the error messaging UI consistent. However, the red border does stand out a bit more. Maybe this is something that can be explored at a later stage @staton-hyse11

**Review by aocenas — APPROVED** [review]

**Comment by aocenas** [comment]
I would also go for version B, but no strong feelings there. The A though feels sort of too "technical", like output from a terminal in IDE or something like that.

**Comment by staton-hyse11** [comment]
I also like option B.  @diegoadams, that's completely right. We can always explore and run A/B tests on things like this to identify what performs best with our users.

**Pull request merged**